runtime_builders/publish_runtime.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/python
+
+# Copyright 2016 Google Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from datetime import datetime
+import json
+import logging
+import os
+from ruamel import yaml
+import subprocess
+import sys
+import tempfile
+
+from google.cloud import storage
+
+# TAG_PATTERN = re.compile('.*/.*/.*:\$\{.*\}')
+
+LANGUAGES = [
+    'java',
+    'python',
+    'ruby',
+    'nodejs',
+    'golang',
+    'dotnet',
+    'php',
+    'test'
+]
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--infile', '-i', help='file(s) to read')
+    parser.add_argument('--bucket', '-b',
+                        help='GCS bucket to publish runtime to',
+                        default='runtime-builders')
+    parser.add_argument('--language', '-l',
```

Comment:
yeah, sorry this is pretty unclear. basically, since this script just takes in a bunch of templated cloudbuild files and then writes them to GCS, it's agnostic to which language runtime each file is associated with, so to make things more explicit I decided to have each run be parameterized with the associated runtime. so a sample run would look like `./publish_runtime.py --infile=cloudbuild.yaml.in --language=python`. but I agree that `runtime-name` is better